src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -205,6 +205,14 @@ private void SetVerb()
                         SetCurlOption(CURLoption.CURLOPT_COPYPOSTFIELDS, string.Empty);
                     }
                 }
+                else
+                {
+                    SetCurlOption(CURLoption.CURLOPT_CUSTOMREQUEST, _requestMessage.Method.Method);
+                    if (_requestMessage.Content != null)
+                    {
+                        SetCurlOption(CURLoption.CURLOPT_UPLOAD, 1L);
```

Comment:
To be honest, I don't remember.  I will be adding more tests here regarding all standard HTTP verbs. 

The RFC specifies that most of the standard verbs can optionally have entity-body. Some standard verbs such as TRACE can never have entity-body:
http://tools.ietf.org/html/rfc7231#section-4.3.8

>A client MUST NOT send a message body in a TRACE request.

And custom HTTP verbs can send entity-body but the server can reject it since it is a 'custom' method with application-defined semantics.